### PR TITLE
Add support for Bravyi-Kitaev transformation in UCCSD ansatz

### PR DESCRIPTION
**Pull request by sabeegrewal**
The UCCSD ansatz uses the HF state as the initial state. This PR prepares that initial state depending on the transformation (JW or BK) used. 

(Note: I confirmed that this works with the H2 VQE workflow.)

### CODE REVIEW

File: src/python/zquantum/vqe/utils.py [hunk 1]
```diff
@@ -1,26 +1,28 @@
...
 ) -> Circuit:
     """Create a circuit corresponding to the exponentiation of an operator. Works only for antihermitian fermionic operators.
 
     Args:
         fermion_generator (openfermion.FermionOperator or 
             openfermion.InteractionOperator): fermionic generator.
-        fermion_transform (str): The name of the qubit to fermion transformation
-            to use.
+        transformation (str): The name of the qubit-to-fermion transformation to use.
+        number_of_qubits (int): The number of qubits in the system. Defaults to None.
```
Comment (mstechly) [inline src/python/zquantum/vqe/utils.py:25]: I would add description of what "None" means.

File: src/python/zquantum/vqe/utils.py [hunk 2]
```diff
@@ -47,3 +49,53 @@ def exponentiate_fermion_operator(
     circuit = exponentiate(qubit_generator)
 
     return Circuit(circuit)
+
+
+def build_hartree_fock_circuit(
```
Comment (mstechly) [inline src/python/zquantum/vqe/utils.py:54]: I don't see tests for this :)

**Comment by sabeegrewal** [comment]
Should be good to go now. I would like either @jromerofontalvo or @jgonthier to take a look before merging though.

**Comment by jgonthier** [comment]
Sorry for the delay! I'm going to have some dinner and then review this code.

**Pull request merged**